### ISSUE TITLE
"Exclude" filter on dashboard breaks queries if you select a value to exclude, and then add it back

### ISSUE THREAD

**Issue by noahmoss** [metabase/metabase#24236]
**To Reproduce**
1. Create a dashboard with a question with a temporal field (like `select * from orders`)
2. Add a filter to the dashboard with the `date/all-options` type
3. Connect the filter to the question and save the dashboard
4. Click on the filter widget > click "Excludes..."  > click any granularity
5. Exclude any values from the filter and click "Update filter"
6. Click on the filter widget again and click "Select all...", then click "Update filter"
7. Card should error

**Logs**

<details>

```
2022-07-22 19:22:08,355 ERROR middleware.catch-exceptions :: Fehlerverarbeitungsanfrage: Don't know how to parse date string "exclude-months-"
{:database_id 1,
 :started_at #t "2022-07-22T19:22:08.252615Z[UTC]",
 :error_type :invalid-parameter,
 :json_query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :query,
  :middleware {:js-int-to-string? true, :ignore-cached-results? false},
  :database 1,
  :query {:source-table 2},
  :parameters
  [{:type :date/all-options, :value "exclude-months-", :target [:dimension [:field 4 nil]], :id "ebaaca9a"}],
  :async? true,
  :cache-ttl nil},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> driver.common.parameters.dates$eval65671$date_string__GT_filter__65676$fn__65677.invoke(dates.clj:399)"
  "driver.common.parameters.dates$eval65671$date_string__GT_filter__65676.invoke(dates.clj:394)"
  "query_processor.middleware.parameters.mbql$eval65873$build_filter_clause__65878$fn__65888.invoke(mbql.clj:57)"
  "query_processor.middleware.parameters.mbql$eval65873$build_filter_clause__65878.invoke(mbql.clj:44)"
  "query_processor.middleware.parameters.mbql$expand.invokeStatic(mbql.clj:86)"
  "query_processor.middleware.parameters.mbql$expand.invoke(mbql.clj:72)"
  "query_processor.middleware.parameters$expand_mbql_params.invokeStatic(parameters.clj:36)"
  "query_processor.middleware.parameters$expand_mbql_params.invoke(parameters.clj:32)"
  "query_processor.middleware.parameters$expand_one.invokeStatic(parameters.clj:48)"
  "query_processor.middleware.parameters$expand_one.invoke(parameters.clj:40)"
  "query_processor.middleware.parameters$expand_all$replace_65982__65983.invoke(parameters.clj:58)"
  "mbql.util.match.impl$replace_in_collection$iter__20513__20517$fn__20518.invoke(impl.cljc:44)"
  "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:43)"
  "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
  "query_processor.middleware.parameters$expand_all$replace_65982__65983.invoke(parameters.clj:58)"
  "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:58)"
  "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:52)"
  "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:55)"
  "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:52)"
  "query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:76)"
  "query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:72)"
  "query_processor.middleware.parameters$eval65998$substitute_parameters_STAR___66003$fn__66004.invoke(parameters.clj:81)"
  "query_processor.middleware.parameters$eval65998$substitute_parameters_STAR___66003.invoke(parameters.clj:78)"
  "query_processor.middleware.parameters$substitute_parameters.invokeStatic(parameters.clj:108)"
  "query_processor.middleware.parameters$substitute_parameters.invoke(parameters.clj:100)"
  "query_processor$preprocess_STAR_$fn__68311.invoke(query_processor.clj:126)"
  "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:124)"
  "query_processor$preprocess_STAR_.invoke(query_processor.clj:119)"
  "query_processor$fn__68319$combined_pre_process__68320$combined_pre_process_STAR___68321.invoke(query_processor.clj:208)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__66291$fn__66296.invoke(resolve_database_and_driver.clj:35)"
  "driver$do_with_driver.invokeStatic(driver.clj:75)"
  "driver$do_with_driver.invoke(driver.clj:71)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__66291.invoke(resolve_database_and_driver.clj:34)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__64861.invoke(fetch_source_query.clj:346)"
  "query_processor.middleware.store$initialize_store$fn__67125$fn__67126.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
  "query_processor.store$do_with_store.invoke(store.clj:38)"
  "query_processor.middleware.store$initialize_store$fn__67125.invoke(store.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__65073.invoke(normalize_query.clj:22)"
  "metabase_enterprise.audit_app.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__67748.invoke(handle_audit_queries.clj:130)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__54189.invoke(constraints.clj:53)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__66164.invoke(process_userland_query.clj:145)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__63713.invoke(catch_exceptions.clj:167)"
  "query_processor.reducible$async_qp$qp_STAR___55800$thunk__55802.invoke(reducible.clj:100)"
  "query_processor.reducible$async_qp$qp_STAR___55800$fn__55804.invoke(reducible.clj:105)"],
 :card_id 1,
 :context :dashboard,
 :error "Don't know how to parse date string \"exclude-months-\"",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data {:type :invalid-parameter, :date-string "exclude-months-"},
 :data {:rows [], :cols []}}
```
</details>

**Expected behavior**
Card should not error

**Screenshots**

https://user-images.githubusercontent.com/32746338/180512948-b46b0138-9e79-46a6-b700-750a8caa944c.mov

**Information about your Metabase Installation:**
Current master

**Severity**
p2?

**Additional context**
Seems to be an issue with `date_filter=exclude-quarters-` in the URL

**Comment by CodeDrivesUs** [comment]
Working on it.

**Comment by KlingerMatheus** [comment]
Taking a look at it.

**Comment by KlingerMatheus** [comment]
Is related to #24235

**Comment by KlingerMatheus** [comment]
> Is related to #24235

@noahmoss take a look at demo on the [PR - Issue #24235](https://github.com/GitStartHQ/client-metabase/pull/106) , this issue was also fixed.

**Comment by gitstart** [comment]
@noahmoss [This](https://github.com/metabase/metabase/pull/24456) PR fixes #24235 as well as this issue
